### ISSUE TITLE
ME0DigiPreRecoProducer: etaResolution cannot be 0.0

### ISSUE THREAD

**Issue by davidlt** [cms-sw/cmssw#16838]
I have noticed that AArch64 IBs RelVals are hanging for a lot of time for the last month or so.

I think, was caused by https://github.com/cms-sw/cmssw/pull/16132 where `etaResolution` was changed to `0.0`.

See `SimMuon/GEMDigitizer/python/muonME0DigisPreReco_cfi.py`

If I look into x86_64, in `ME0PreRecoGaussianModel::ME0PreRecoGaussianModel` we set 
```
24        sigma_v(config.getParameter<double>("etaResolution")),
(gdb) n
(gdb) p sigma_v
$10 = 0
```

But in `ME0PreRecoGaussianModel::simulateNoise` we get division by zero:
```
116       double heightIt = sigma_v;
(gdb)
117       int heightbins  = height/heightIt; // round down
(gdb) p heightIt
$13 = 0
(gdb) p height
$14 = 12.119999885559082
(gdb) n
119       edm::LogVerbatim("ME0PreRecoGaussianModelNoise") << "[ME0PreRecoDigi :: sNoise]["<<roll->id().rawId()<<"] :: roll with id = "<<roll->id();
(gdb) p heightbins
$15 = -2147483648
```

On x86_64 we get negative number (MIN_INT) after division, but on aarch64 we get large positive number (MAX_INT):
```
125       for(int hx=0; hx<heightbins; ++hx) {
(gdb) p heightbins
$1 = 2147483647
(gdb) p height
$2 = 12.119999885559082
(gdb) p heightIt
$3 = 0
(gdb) p sigma_v
$4 = 0
```

Thus we are stuck in the cycle for a very long time. This seems to happen in a lot of workflows, thus plugging all cores from doing anything else.

Thus `etaResolution` cannot be set to zero.

**Comment by davidlt** [comment]
assign simulation

**Comment by davidlt** [comment]
I can confirm that reverting one single live made workflow working again:
```
diff --git a/SimMuon/GEMDigitizer/python/muonME0DigisPreReco_cfi.py b/SimMuon/GEMDigitizer/python/muonME0DigisPreReco_cfi.py
index 8dd34b8..9bbbd45 100644
--- a/SimMuon/GEMDigitizer/python/muonME0DigisPreReco_cfi.py
+++ b/SimMuon/GEMDigitizer/python/muonME0DigisPreReco_cfi.py
@@ -6,7 +6,7 @@ simMuonME0Digis = cms.EDProducer("ME0DigiPreRecoProducer",
     digiPreRecoModelString = cms.string('PreRecoGaussian'),
     timeResolution = cms.double(0.0), # in ns
     phiResolution = cms.double(0.03), # in cm average resolution along local x in case of no correlation
-    etaResolution = cms.double(0.0),  # in cm average resolution along local y in case of no correlation
+    etaResolution = cms.double(1.0),  # in cm average resolution along local y in case of no correlation
     constantPhiSpatialResolution = cms.bool(True),
     useCorrelation = cms.bool(False),
     useEtaProjectiveGEO = cms.bool(False),
```

Someone needs to `etaResolution` which is non zero, or rewrite the code.

**Comment by davidlt** [comment]
Fotgot to mention, for testing I used 23646.0 step2_DIGI_L1_DIGI2RAW_HLT.py

**Comment by smuzaffar** [comment]
assign simulation

**Comment by civanch** [comment]
@calabria , @kpedro88 ,
likely a protection against division by zero due to etaResolution should be added.

**Comment by kpedro88** [comment]
@civanch this was just fixed by @jshlee: https://github.com/cms-sw/cmssw/pull/16605/commits/4cd46d74530407f33da4b522c2ab54f5b39ae58b

**Comment by civanch** [comment]
+1
Fixed in #16605
